runtime/doc/lua.txt

```diff
@@ -4433,4 +4433,160 @@ tohtml.tohtml({winid}, {opt})                         *tohtml.tohtml.tohtml()*
+                    vim.net.download("https://httpbingo.org/anything", {
+                      follow_redirects = true,
+                    })
+
+                    -- Download a file while sending headers
+                    vim.net.download("https://httpbingo.org/anything", {
+                      headers = {
+                        Authorization = { "Bearer foo" },
+                      },
+                    })
+
+                    -- Download a file while handling basic auth
+                    vim.net.download("https://httpbingo.org/basic-auth/user/password", {
+                      credentials = "user:password",
+                    })
+
+                    -- Download a file without overriding a previous file with the same name
+                    vim.net.download("https://httpbingo.org/anything", {
+                      override = false,
+                    })
+<
+                • {download_location}? (`string`) Path to write the downloaded
+                  file to. If not provided, the one inferred form the URL will
+                  be used. Defaults to `nil`
+                • {try_suggested_remote_name}? (`boolean`) Whether the
+                  `Content-Disposition` response header should be taken into
+                  account to decide the name of the downloaded file. Fallbacks
+                  to `download_location`. Defaults to `false`
+                • {credentials}? (`string`) Credentials with the format
+                  `username:password`. Defaults to `nil`
+                • {override}? (`boolean`) Whether the file should be
+                  overridden if it already exists. Defaults to `true`
+                • {remove_leftover_on_error}? (`boolean`) Whether the file
+                  should be removed if an error happens while downloading it.
+                  Defaults to `false`
+                • {compressed}? (`boolean`) Whether the file should be
+                  requested in a compressed format (and decompressed
+                  automatically). Defaults to `false`
+                • {max_filesize}? (`integer`) Maximum size in bytes of the
+                  file to downlaod. The download will fail if the file
+                  requested is larger. Defaults to `nil`
+                • {raw}? (`boolean`) Disables all internal HTTP decoding of
+                  content or transfer encodings. Unaltered, raw, data is
+                  passed. Defaults to `false`
+                • {headers}? (`table`) string[]> Request headers. Defaults to
+                  `nil`
+                • {proxy}? (`string`) Proxy URL in the format
+                  `scheme ":" ["//" authority] path ["?" query]` where
+                  authority follows the format
+                  `[userinfo "@"] host [":" port]`. Defaults to `nil`
+                • {proxy_credentials}? (`string`) Proxy credentials with the
+                  format `username:password`. Defaults to `nil`
```

Comment:
I think we could use a table option (or `string|table`) instead: `proxy = { url = 'foo', credentials = 'bar:baz }`